examples/platform_channel/macos/Runner/PowerSource.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import IOKit.ps
+
+enum PowerState {
+  case ac
+  case battery
+  case unknown
+}
+
+/// A convenience wrapper for an IOKit power source.
+class PowerSource {
+  let info = IOPSCopyPowerSourcesInfo().takeRetainedValue()
+  lazy var sources = IOPSCopyPowerSourcesList(info).takeRetainedValue() as Array
+
+  /// Returns the current power source capacity. Apple-defined power sources will return this value
+  /// as a percentage.
+  func getCurrentCapacity() -> Int {
+    if !sources.isEmpty {
+      let source = sources[0]
```

Comment:
Optional nit: I believe (I don't do a lot of Swift and this is freehand) you can make these two lines just:

```
if let source = sources.first {
```

(NSArray `first` returns nil rather than throwing for empty, unlike Dart, since it can't contain nil.)